pkg/config/analysis/analyzers/util/service_lookup.go

```diff
@@ -20,15 +20,32 @@ import (
 	corev1 "k8s.io/api/core/v1"
 
 	"istio.io/api/annotation"
+	meshconfig "istio.io/api/mesh/v1alpha1"
 	"istio.io/api/networking/v1alpha3"
 	"istio.io/istio/pkg/config/analysis"
 	"istio.io/istio/pkg/config/resource"
 	"istio.io/istio/pkg/config/schema/gvk"
 )
 
+func GetCustomClusterDomain(ctx analysis.Context) string {
+	cusClusterDomain := ""
+	// use meshConfig.trustDomain changed the default domain
```

Comment:
trustDomain and cluster domain are 100% orthogonal to each other. There is absolutely no relationship **at all** (beyond the same default) and they should not be used interchange-ably like this

Reply:
Let me try to find a better solution 🤔